preprocessing/nextclade/src/loculus_preprocessing/config.py

```diff
@@ -91,6 +91,8 @@ def get_config() -> Config:
     config = Config()
 
     # Overwrite config with config in config_file
+    if config_file:
```

Comment:
I don't quite understand the change here - can you explain? Who passes the config_file arg? As opposed to via `args.config_file`?

Reply:
config_file will only be passed by the test at this point, I didn't find a smart way to mock `args.config_file`. `args.config_file` is passed as a run time argument and adding this in a test where I am not running the package isn't trivial (maybe you know how to do this I couldn't find a smart way)

So I decided to just add config_file as an optional argument - this makes sure that the config used by the test is parsed in the same way it would be when the package is called via the command line, additionally when run via the command line args.config_file will overwrite config_file making sure this change can't break anything.

Reply:
Thanks for the explanation! I'll quickly think about this, and if I can't figure it out add a comment so at least we know why it exists as an arg :)

Reply:
Ok, this pattern with a test only arg that's optional is not so bad - I've just added a comment and refactored this function in general to clean up a bit (not just related to your addition)

Reply:
See #2973